AsyncDisplayKit/ASVideoNode.mm

```diff
@@ -171,8 +176,58 @@ - (void)didLoad
           _placeholderImageNode.frame = self.bounds;
           [self insertSubnode:_placeholderImageNode atIndex:0];
         });
-      }];
-    });
+      }
+    }];
+  });
+}
+
+- (void)interfaceStateDidChange:(ASInterfaceState)newState fromState:(ASInterfaceState)oldState
+{
+  [super interfaceStateDidChange:newState fromState:oldState];
+	
+	BOOL nowVisible = ASInterfaceStateIncludesVisible(newState);
+	BOOL wasVisible = ASInterfaceStateIncludesVisible(oldState);
+
+  if (!nowVisible) {
+    if (wasVisible) {
+      if (_shouldBePlaying) {
+        [self pause];
+        _shouldBePlaying = YES;
```

Comment:
This is not necessary, unless the condition is supposed to be !_shouldBePlaying?